codersdk/deployment_test.go

```diff
@@ -568,3 +568,69 @@ func TestPremiumSuperSet(t *testing.T) {
 	require.NotContains(t, enterprise.Features(), "", "enterprise should not contain empty string")
 	require.NotContains(t, premium.Features(), "", "premium should not contain empty string")
 }
+
+func TestNotificationsCanBeDisabled(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                       string
+		expectNotificationsEnabled bool
+		environment                []serpent.EnvVar
+	}{
+		{
+			name:                       "NoDeliveryMethodSet",
+			environment:                []serpent.EnvVar{},
+			expectNotificationsEnabled: false,
+		},
+		{
+			name: "SMTP_DeliveryMethodSet",
+			environment: []serpent.EnvVar{
+				{
+					Name:  "CODER_EMAIL_SMARTHOST",
+					Value: "localhost:587",
+				},
+			},
+			expectNotificationsEnabled: true,
+		},
+		{
+			name: "Webhook_DeliveryMethodSet",
+			environment: []serpent.EnvVar{
+				{
+					Name:  "CODER_NOTIFICATIONS_WEBHOOK_ENDPOINT",
+					Value: "https://example.com/webhook",
+				},
+			},
+			expectNotificationsEnabled: true,
+		},
+		{
+			name: "WebhookAndSMTP_DeliveryMethodSet",
+			environment: []serpent.EnvVar{
+				{
+					Name:  "CODER_NOTIFICATIONS_WEBHOOK_ENDPOINT",
+					Value: "https://example.com/webhook",
+				},
+				{
+					Name:  "CODER_EMAIL_SMARTHOST",
+					Value: "localhost:587",
+				},
+			},
+			expectNotificationsEnabled: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
BTW this is no longer necessary [as of Go v1.22](https://go.dev/blog/loopvar-preview).

Reply:
Should I add a linter exception for this so it doesn't complain?

Reply:
That'd be wonderful; in a separate PR though - let's land this one